src/main.go

```diff
@@ -31,6 +32,44 @@ func stderr(f string, a ...interface{}) {
 	fmt.Fprintln(os.Stderr, strings.TrimSuffix(out, "\n"))
 }
 
+// hasUnrecognizedKeys finds unrecognized keys and warns about them on stderr.
+// returns false when no unrecognized keys were found, true otherwise.
+func hasUnrecognizedKeys(inCfg interface{}, refType reflect.Type) (warnings bool) {
+
+	switch inCfg.(type) {
+	case map[interface{}]interface{}:
+		ks := inCfg.(map[interface{}]interface{})
+		for key := range ks {
+			i := 0
+			sf := reflect.StructField{}
+			for ; i < refType.NumField(); i++ {
+				sf = refType.Field(i)
+				tv := sf.Tag.Get("yaml")
+				if tv == key {
+					if warn := hasUnrecognizedKeys(ks[key], sf.Type); warn {
+						warnings = true
+					}
+					break
```

Comment:
If you continue to the outer loop, you can narrow the scope of `i` and remove the condition on the warning.